### PR TITLE
Add kairos test utils

### PR DESCRIPTION
**Pull request by marijanp**

### CODE REVIEW

File: kairos-test-utils/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+[lib]
+
+[dependencies]
+
+log = "*"
+tokio = { version = "*", features = [ "full", "macros" ] }
+env_logger = "*"
```
Comment (Avi-D-coder) [inline kairos-test-utils/Cargo.toml:13]: I'd prefer to uniformly use `tracing` over `log`/`env_logger`.
This can be changed in a followup eventually.

**Review by Avi-D-coder — APPROVED** [review]

File: Cargo.lock [hunk 2]
```diff
@@ -407,50 +526,66 @@ checksum = "baf1de4339761588bc0619e3cbc0120ee582ebb74b53b4efbf79117bd2da40fd"
 
 [[package]]
 name = "clap"
-version = "4.4.18"
+version = "4.5.3"
```
Comment (marijanp) [inline Cargo.lock:529]: @koxu1996

Comment (koxu1996) [reply Cargo.lock:529]: It was already updated to the same version [here](https://github.com/cspr-rad/kairos/pull/36/files#diff-13ee4b2252c9e516a0547f2891aa2105c3ca71c6d7a1e682c69be97998dfc87eL410), and there was no warning in CI/CD (there is no warning when building locally as well).

File: kairos-test-utils/Cargo.toml (hunk 1 above)
Comment (marijanp) [reply kairos-test-utils/Cargo.toml]: replaced log with tracing

**Review by Avi-D-coder — APPROVED** [review]

**Comment by asladeofgreen** [comment]
General comment: as stated via Zulip DM we could factor out this work into a dedicated `cctl-rs` library within the `casper-network` GH space.  To be discussed in more detail a little down the road.

File: kairos-server/Cargo.toml [hunk 3]
```diff
@@ -18,7 +18,6 @@ axum-extra = { version = "0.9", features = [
     "typed-header",
     "json-deserializer",
 ] }
-thiserror = "1"
```
Comment (koxu1996) [inline kairos-server/Cargo.toml:21]: I would leave it as it is.

File: kairos-test-utils/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+[dependencies]
+
+tokio = { version = "*", features = [ "full", "tracing", "macros" ] }
+tempfile = "*"
+tracing = "0.1"
+nom = "*"
+
+[dev-dependencies]
+casper-client = "2"
+anyhow = "*"
```
Comment (koxu1996) [inline kairos-test-utils/Cargo.toml:18]: Instead of wildcards, I would rather see version specification.

File: kairos-test-utils/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+[package]
```
Comment (koxu1996) [inline kairos-test-utils/Cargo.toml:1]: License field will be necessary after synchronizing base branch.

File: kairos-test-utils/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+name = "kairos-test-utils"
+version = "0.1.0"
+edition = "2021"
```
Comment (koxu1996) [inline kairos-test-utils/Cargo.toml:4]: I noticed that all packages (excluding prover) are inheriting from the workspace:

```
version.workspace = true
edition.workspace = true
```

Should it be the same case here?

File: kairos-test-utils/src/cctl.rs [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+                    panic!("Can't find ports for node with id {}", node_id)
+                }
+            })
+            .collect();
+
+        tracing::info!("Waiting for network to start processing blocks");
+        std::thread::sleep(std::time::Duration::from_secs(5));
+        let output = Command::new("cctl-chain-await-until-block-n")
+            .env("CCTL_ASSETS", &assets_dir)
+            .arg("height=0")
```
Comment (koxu1996) [inline kairos-test-utils/src/cctl.rs:87]: I have not tested it, but maybe we should wait for `height=1` (genesis block)?

File: kairos-test-utils/src/cctl.rs [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+                        port,
+                    }
+                } else {
+                    panic!("Can't find ports for node with id {}", node_id)
+                }
+            })
+            .collect();
+
+        tracing::info!("Waiting for network to start processing blocks");
+        std::thread::sleep(std::time::Duration::from_secs(5));
```
Comment (koxu1996) [inline kairos-test-utils/src/cctl.rs:84]: I would just rely on `cctl-chain-await-until-block-n`.

```suggestion
```

**Comment by koxu1996** [comment]
@marijanp Looks good, few minor improvements to be done. I think it should be noted that this test util has implicit `bash` dependency.

File: kairos-test-utils/Cargo.toml (hunk 4 above)
Comment (marijanp) [reply kairos-test-utils/Cargo.toml]: done https://github.com/cspr-rad/kairos/pull/34/commits/acd30355069f13f1194ec3468daaa9736656f00b

File: kairos-test-utils/Cargo.toml (hunk 5 above)
Comment (marijanp) [reply kairos-test-utils/Cargo.toml:1]: done https://github.com/cspr-rad/kairos/pull/34/commits/4276b0b3b7d0937e4755d040a8fc8c1e22562521

File: kairos-test-utils/Cargo.toml (hunk 6 above)
Comment (marijanp) [reply kairos-test-utils/Cargo.toml]: done https://github.com/cspr-rad/kairos/pull/34/commits/4276b0b3b7d0937e4755d040a8fc8c1e22562521

File: kairos-test-utils/src/cctl.rs (hunk 7 above)
Comment (marijanp) [reply kairos-test-utils/src/cctl.rs]: replaced in favor of querying the rector_state of the node to determine readyness https://github.com/cspr-rad/kairos/pull/34/commits/acd30355069f13f1194ec3468daaa9736656f00b

File: kairos-test-utils/src/cctl.rs (hunk 8 above)
Comment (marijanp) [reply kairos-test-utils/src/cctl.rs]: replaced in favor of querying the rector_state of the node to determine readyness https://github.com/cspr-rad/kairos/pull/34/commits/acd30355069f13f1194ec3468daaa9736656f00b

File: Cargo.lock (hunk 2 above)
Comment (koxu1996) [reply Cargo.lock]: Okay, I see no clap update now :relieved:.

File: kairos-tx/Cargo.toml
Comment (koxu1996) [inline kairos-tx/Cargo.toml:1]: Those changes should not be part of this PR.

**Review by koxu1996 — APPROVED** [review]
